php/libraries/NDB_Page.class.inc

```diff
@@ -927,6 +927,110 @@ class NDB_Page implements RequestHandlerInterface
         );
     }
 
+    /**
+     * Get the list of projects for which the current user has permission
+     * To populate the module projects filter options
+     *
+     * @param bool $numeric_keys If the returned projects keys
+     *                           should be numeric (ProjectID)
+     *                           or strings (project name)
+     *
+     * @return array - an associative array of the form (ID => name)
+     */
+    public function getProjectOptions(
+        bool $numeric_keys = true
+    ): array {
+        $projectsOptions = [];
+        $userProjectsIDs = \NDB_Factory::singleton()->user()->getProjectIDs();
+        foreach ($userProjectsIDs as $projectID) {
+            $projectsOptions[strval($projectID)]
+                = \Project::getProjectFromID($projectID)->getName();
+        }
+
+        if (!$numeric_keys) {
+            foreach ($projectsOptions as $id => $name) {
+                unset($projectsOptions[$id]);
+                $projectsOptions[$name] = $name;
+            }
+        }
+
+        return $projectsOptions;
+    }
+
+    /**
+     * Get the list of subprojects for which the current user has permission
+     * To populate the module subprojects filter options
+     *
+     * @param bool $numeric_keys If the returned subprojects keys
+     *                           should be numeric (SubprojectID)
+     *                           or strings (subproject name)
+     *
+     * @return array - an associative array of the form (ID => name)
+     */
+    public function getSubprojectOptions(
```

Comment:
Similarly to User projects, Subprojects don't seem to be related to pages. If the user had a `getProjects` method this would just be an `array_reduce` on `$user->getProjects()`